### PR TITLE
Additional volume options

### LINKED ISSUE #155: Allow IOPS config option for io1 volumeType
**Is this a BUG REPORT or FEATURE REQUEST?**:
FEATURE
**What happened**:
Allow IOPS config option for io1 volumeType 

https://github.com/keikoproj/instance-manager/blob/5843c10e7b365f7206542e08efd754413d51799c/api/v1alpha1/instancegroup_types.go#L193-L196

More Details about io1
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-volume-types.html

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

- Kubernetes version:

```bash
kubectl version -o yaml
```

**Other debugging information (if applicable)**:

- InstanceGroup status:

```bash
kubectl describe instancegroup <ig-name>
```

- controller logs:

```bash
kubectl logs <instance-manager pod>
```

### PR DESCRIPTION
**Pull request by eytan-avisror**
Fixes #155 

Adds additional volume options to NodeVolumes

### CODE REVIEW

File: controllers/providers/aws/aws.go [hunk 1]
```diff
@@ -151,15 +153,32 @@ func (w *AwsWorker) InstanceProfileExist(name string) (*iam.InstanceProfile, boo
...
+	if delete != nil {
+		device.Ebs.DeleteOnTermination = delete
+	} else {
+		device.Ebs.DeleteOnTermination = aws.Bool(true)
+	}
+	if encrypt != nil {
+		device.Ebs.Encrypted = encrypt
+	}
+	if iops != 0 {
+		device.Ebs.Iops = aws.Int64(iops)
```
Comment (vgunapati) [inline controllers/providers/aws/aws.go:172]: can we add this validation https://github.com/aws/aws-sdk-go/blob/v1.33.16/service/autoscaling/api.go#L9756

Comment (eytan-avisror) [reply controllers/providers/aws/aws.go:172]: I can add a validation that IOPS is not 1-99

**Comment by eytan-avisror** [comment]
> This feature is to gather IOPS, DeleteonTermination and Encrypted info from AWS right ?

It's to allow these fields to be configurable in the custom resource

File: controllers/providers/aws/aws.go (hunk 1 above)
Comment (eytan-avisror) [reply controllers/providers/aws/aws.go:172]: added

**Comment by shaoxt** [comment]
OK, what if the old configuration doesn't have UserDataStage/data or UserDataStage/stage ?
Will it causes failure or the validation could take care of the case ?

> > This feature is to gather IOPS, DeleteonTermination and Encrypted info from AWS right ?
> 
> It's to allow these fields to be configurable in the custom resource

**Comment by eytan-avisror** [comment]
> OK, what if the old configuration doesn't have UserDataStage/data or UserDataStage/stage ?
> Will it causes failure or the validation could take care of the case ?
> 
> > > This feature is to gather IOPS, DeleteonTermination and Encrypted info from AWS right ?
> > 
> > 
> > It's to allow these fields to be configurable in the custom resource

The UserDataStage type changes are backwards compatible, we only added a "name" field and this PR also makes it 'omitempty' to make sure if an old CR did not have this field it won't break.
The name field we added is for convenience any way and is not mandatory

**Pull request merged**